ompi/communicator/comm_cid.c

```diff
@@ -771,9 +771,29 @@ static int ompi_comm_nextcid_check_flag (ompi_comm_request_t *request)
 /* Non-blocking version of ompi_comm_activate */
 static int ompi_comm_activate_nb_complete (ompi_comm_request_t *request);
 
-static int ompi_comm_activate_complete (ompi_communicator_t **newcomm, ompi_communicator_t *comm)
+/* Callback function to set comunicator disjointness flags */
+static inline void ompi_comm_set_disjointness_nb_complete (ompi_comm_cid_context_t *context) {
+    if (!OMPI_COMM_IS_DISJOINT_SET(*context->newcommp)) {
```

Comment:
This function should only be called once per communicator. Calling a second time should trigger an error, at least a message in debug mode.

Reply:
Thanks I will think about that.

Reply:
Please see update - I decided to add a warning message instead of hard failure.